apps/backend/listing/src/createListing.ts

```diff
@@ -0,0 +1,80 @@
+import { Request, Response, NextFunction } from "express";
+import { IDatabase } from "pg-promise";
+
+// POST /api/listing - Create a new listing
+const createListing = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+  db: IDatabase<object>,
+) => {
+  // TODO: AUTHENTICATION
+  const userID = 1;
+  const { listing } = req.body;
+
+  if (!listing) {
+    return res.status(400).json({ error: "missing parameter in request" });
+  }
+
+  const { title, description, price, location, images } = listing;
+
+  if (!title || !description || !price || !location || !images) {
+    return res.status(400).json({ error: "missing parameter in request" });
+  }
+
+  if (!location.latitude || !location.longitude) {
+    return res.status(400).json({ error: "missing parameter in request" });
+  }
+
+  const formattedLocation = `(${location.latitude},${location.longitude})`;
+
+  try {
+    const createdListing = await db.one(
+      `INSERT INTO listings (title, description, price, location, image_urls, status, created_at, modified_at, seller_id)
+       VALUES ($1, $2, $3, $4, $5, 'AVAILABLE', NOW(), NOW(), $6)
+       RETURNING *`,
+      [
+        title,
+        description,
+        price,
+        formattedLocation,
+        images.map((image: { url: string }) => image.url),
+        userID,
+      ],
+    );
+
+    const sellerProfile = await db.one(
+      `SELECT user_id AS "userID", username, name, bio, profile_pic_url AS "profilePictureUrl"
+       FROM users
+       WHERE user_id = $1`,
+      [userID],
+    );
+
+    const responseListing = {
+      listingID: createdListing.listing_id,
+      seller_profile: sellerProfile,
+      title: createdListing.title,
+      description: createdListing.description,
+      price: createdListing.price,
+      location: {
+        latitude: parseFloat(
+          createdListing.location.split(",")[0].replace("(", ""),
```

Comment:
this formatting should be different or skipped altogether since we should have the lat/long variables already